src/App/VPSScan/Scan/Core.hs

```diff
@@ -94,3 +128,42 @@ getSherlockInfo FossaOpts{..} = runHTTP $ do
   (baseUrl, baseOptions) <- parseUri fossaUrl
   resp <- req GET (sherlockInfoEndpoint baseUrl) NoReqBody jsonResponse (baseOptions <> header "Content-Type" "application/json" <> auth)
   pure (responseBody resp)
+
+getProjectScanFilters :: (Has (Lift IO) sig m, Has Diagnostics sig m) => FossaOpts -> Locator -> m CoreFilterExpressions
+getProjectScanFilters FossaOpts{..} locator = runHTTP $ do
+  let auth = coreAuthHeader fossaApiKey
+
+  (baseUrl, baseOptions) <- parseUri fossaUrl
+  resp <- req GET (projectScanFiltersEndpoint baseUrl locator) NoReqBody jsonResponse (baseOptions <> header "Content-Type" "application/json" <> auth)
+  pure (responseBody resp)
+
+overrideScanFilters :: (Has (Lift IO) sig m, Has Diagnostics sig m, Has Trace sig m) => VPSOpts -> Locator -> m (VPSOpts, Bool)
```

Comment:
I'm not sure what our preferred syntax for something like this is, but I've always enjoyed the "multiple function definition" approach rather than using case statements. It's a little complicated in this case as you have to pattern match into a type in a record, but I think it works out nicely.

Something like this. This is untested but VS Code is not complaining about it:

```haskell
overrideScanFilters :: (Has (Lift IO) sig m, Has Diagnostics sig m, Has Trace sig m) => VPSOpts -> Locator -> m (VPSOpts, Bool)
overrideScanFilters vpsOpts@VPSOpts {filterBlob = (FilterExpressions "[]")} locator = do
    trace "[All] Fetching scan file filter from FOSSA"
    overrideFilters <- getProjectScanFilters (fossa vpsOpts) locator
    let overrideFilterBlob = toFilterExpressions overrideFilters
    trace $ unpack $ "[All] Using filter: " <> (unFilterExpressions overrideFilterBlob)
    pure (vpsOpts{filterBlob = overrideFilterBlob}, True)

overrideScanFilters vpsOpts _ = do
    trace "[All] Scan file filters provided locally"
    pure (vpsOpts, False)

```